src/openms/include/OpenMS/CONCEPT/Colorizer.h

```diff
@@ -0,0 +1,340 @@
+// ETH Zurich, and Freie Universitaet Berlin 2002-2022.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Moritz Berger, Tetana Krymovska$
+// $Authors: Moritz Berger, Tetana Krymovska$
+// --------------------------------------------------------------------------
+
+#pragma once
+
+//#include <OpenMS/KERNEL/MSExperiment.h>
+
+#include <array>
+#include <iosfwd>
+#include <sstream>
+#include <OpenMS/config.h>
+// #include </buffer/ag_bsc/pmsb_22/tetak94/openms_colorizerT.2/openms/src/openms/include/OpenMS/config.h.in>
+
+
+namespace OpenMS
+{
+  /// enum COLOR for easier Object initialisation.
+  enum class COLOR
+  {
+    black,
+    red,
+    green,
+    yellow,
+    blue,
+    magenta,
+    cyan,
+    white,
+    RESET, ///< reset the color to the previous (default?) color
+  };
+
+  /**
+   * @brief A class, that provides options for colored output with the "<<" operator for output streams (cout, cerr)
+   *
+   */
+
+
+
+  class OPENMS_DLLAPI Colorizer
+  {
+public:
+    /// Constructor
+    Colorizer(const COLOR color);
+
+    /// Copy constructor
+    // Colorizer(const Colorizer &rhs);
+
+    ///Assignment Operator
+  
+
+    /// Destructor
+    ~Colorizer();
+
+    ///
+    void outputToStream(std::ostream& o_stream);
+
+    ///
+    void colorStream(std::ostream& stream) const;
+
+    ///
+    void resetColor(std::ostream& stream);
```

Comment:
these functions can be `protected` - not `public`.
(renaming by adding `_` at the end, e..g `outputToStream_`

Reply:
however, it would be good to have a `Colorizer::reset()` function, which can be used like this:
`std::cout << blue() << " all is blue " << 5 < " still blue " << blue.reset() << " not blue anymore";`

implementation:
```
Colorizer& reset()
{
   this->input_.str(""); // clear the stringstream
   reset_ = true;  // reset color after use
}
```

Reply:
using another colorizer object (reset_color) to reset the color is actually not that ideal, e.g.
`std::cout << blue() << " all is blue " << 5 < " still blue " << reset_color() << " not blue anymore";

So please delete that object.